### PR TITLE
Fixes and tweaks

### LINKED ISSUE #17631: Disinfecting with suit storage units allows 'teleportation' of things.
[Note]: # (State what the issue is from a "whats wrong" prospective. Issue reports should clearly allow maintainers to understand whats wrong and how to test/reproduce if that is not obvious. Avoid ambiguity. Start your issue report below both of these lines (or remove them))

[Admins]: # (If you are reporting a bug that occured AFTER you used varedit/admin buttons to alter an object out of normal operating conditions, please verify that you can re-create the bug without the varedit usage/admin buttons before reporting the issue.)

Put hard suit in a suit storage unit, run disinfect. It will spit out the equipment. Put it on. Open suit storage unit, it says the suit is still inside. Click it, the suit suddenly teleports off of you and onto the space.

You could use this to trick security into putting on disinfected suits and then teleport their suits off them once they're in space.

### LINKED ISSUE #18102: The nar'sie rune can be scribed in space
Make a floating room in space, pressurise it, put a tp rune inside.

Scribe summon nar'sie rune, crew is told the location is Space. Crew does not find your secret cult box, Narnar ascends.

Bug or feature?

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
rscdel: Removed the global message played when Nar-Sie _begins_ to spawn (but not when it actually spawns).
tweak: Drunkenness recovery speed now increases with how drunk the imbiber is and is much quicker when the imbiber is asleep.
tweak: Suit storage units now take three seconds to enter (up from one) and have different sounds and messages for UV ray cauterization.
bugfix: Fixed some bugs with the suit storage unit, inserting mobs, and contents to seemed to duplicate themselves.
bugfix: The Summon Nar-Sie rune can now only be drawn on original station tiles and fails to invoke if scribed on the station then moved elsewhere.
/ :cl:
- Fixes #17631 
- Fixes #18102

### CODE REVIEW

**Comment by Xhuis** [comment]
With the sheer amount of people that have hounded me to make this change for a very long time, you might want to consider speedmerging if it isn't too much of a problem.

**Comment by phil235** [comment]
If I'm reading the code correctly this means you need to drink 400u of beer to even feel slightly tispy.

**Comment by Xhuis** [comment]
My math told me that 50u of beer would give you apx. 20 drunkenness. I haven't had time to test it in-game but it should be relatively simple.

**Comment by phil235** [comment]
`H.drunkenness = max((H.drunkenness + (sqrt(volume) * boozepwr * ALCOHOL_RATE)), 0)`

H.drunkenness = max((H.drunkenness + (sqrt(50) * 25 * 0.001)), 0) //new numbers for 50u beer.
sqrt(50) * 25 * 0.001 = 0.175 which is lower than 0.5 of the natural decrease, so you won't get drunk at all.

**Comment by Xhuis** [comment]
I didn't factor in the decrease. If I just changed the passive drunkenness decrease and kept the rate as-is before this PR, would it still accomplish the effect of toning down drunkenness speed?

**Comment by phil235** [comment]
It would help, but I still think a system where the natural decrease varies with our drunkenness level would allow players to be tipsy or drunk more easily without risking dying from drunkenness.

**Comment by Xhuis** [comment]
I'll see about implementing it. I'm also going to bundle a few more tweaks (mainly optimizations) and fixes into this PR.

**Comment by Xhuis** [comment]
All right. I changed the formulas to `drunkenness / 25` when the imbiber is awake and `drunkenness / 10` when they're asleep. I also added in some suit storage unit fixes, tweaks, and optimizations - mainly, preventing some runtime errors and strange object behavior.

**Comment by Xhuis** [comment]
I added LOOC, taken from other servers. It's essentially OOC that only functions within 7 tiles. Currently, it cannot be used while dead, and admins can see all LOOC that occurs. _I might relocate this to a different pull request, or remove it entirely if the maintainers don't want it._

**Comment by ChangelingRain** [comment]
ATOMIZE YOUR PRS YA SHIT

**Comment by Jordie0608** [comment]
LOOC was proposed before and turned down each time thus far

**Comment by Xhuis** [comment]
Oh, it was? Ok, I'll cut it out
@ChangelingRain I DON'T KNOW WHAT THAT MEANS

File: code/game/machinery/suit_storage_unit.dm [hunk 1]
```diff
@@ -219,10 +226,15 @@
 			// The wires get damaged too.
```
Comment (phil235) [inline code/game/machinery/suit_storage_unit.dm]: can you fix a little thing above this? the code is calling qdel() for each item inside but it nullifies the reference beforehand so it currently doesn't qdel() anything. It should be qdel(X) and THEN X = null

**Comment by phil235** [comment]
add Fixes #xxxx for the SSU fix.

File: code/game/machinery/suit_storage_unit.dm (hunk 1 above)
Comment (Xhuis) [inline code/game/machinery/suit_storage_unit.dm]: Nice. Yeah, I'll do that in a bit.

**Comment by Xhuis** [comment]
@phil235 If there's an issue report, I didn't see it. The issues I fixed here were discovered by myself and unreported.

**Comment by Xhuis** [comment]
Fixed #18102 and modified the Nar-Sie rune a bit (check PR body for details).

File: code/game/gamemodes/cult/ritual.dm [hunk 2]
```diff
@@ -214,6 +214,10 @@ This file contains the arcane tome files.
 			else if(!cult_mode.eldergod)
 				user << "<span class='cultlarge'>\"I am already here. There is no need to try to summon me now.\"</span>"
 				return
+			var/area/AR = get_area(user)
+			if(initial(AR.name) == "Space" || istype(Turf, /turf/open/space) || user.z != ZLEVEL_STATION)
+				user << "<span class='warning'>The rune cannot be scribed this far from the station!</span>"
+				return
 			var/confirm_final = alert(user, "This is the FINAL step to summon Nar-Sie, it is a long, painful ritual and the crew will be alerted to your presence", "Are you prepared for the final battle?", "My life for Nar-Sie!", "No")
```
Comment (ChangelingRain) [inline code/game/gamemodes/cult/ritual.dm]: You need to check again after the alert.

File: code/game/gamemodes/cult/ritual.dm [hunk 3]
```diff
@@ -214,6 +214,10 @@ This file contains the arcane tome files.
 			else if(!cult_mode.eldergod)
 				user << "<span class='cultlarge'>\"I am already here. There is no need to try to summon me now.\"</span>"
 				return
+			var/area/AR = get_area(user)
+			if(initial(AR.name) == "Space" || istype(Turf, /turf/open/space) || user.z != ZLEVEL_STATION)
```
Comment (ChangelingRain) [inline code/game/gamemodes/cult/ritual.dm]: The `istype(Turf, /turf/open/space)` should be for all runes tbh

**Comment by Xhuis** [comment]
I think Travis has gone insane and doesn't want to check this PR. Nonetheless, I'll get those issues you outlined fixed shortly.

**Comment by Xhuis** [comment]
All right. Runes can't be scribed in space anymore, and runes for summoning Nar-Sie have to be on original station turf (other runes can be off-station).

**Comment by optimumtact** [comment]
Conflicts

**Comment by Xhuis** [comment]
### GOD
## FUCKING
# DAMN IT

**Pull request merged**